simple-generated/src/event.rs

```diff
@@ -1,87 +1,116 @@
-use crate::app::AppResult;
-use ratatui::crossterm::event::{
-    self, Event as CrosstermEvent, KeyEvent, KeyEventKind, MouseEvent,
-};
+use crossterm::event::{self, Event as CrosstermEvent, KeyEvent, KeyEventKind, MouseEvent};
 use std::sync::mpsc;
 use std::thread;
 use std::time::{Duration, Instant};
 
 /// Terminal events.
-#[derive(Clone, Copy, Debug)]
+#[derive(Debug)]
 pub enum Event {
-    /// Terminal tick.
+    /// The tick event is sent at a regular interval and can be used to trigger application state
+    /// updates or animations.
     Tick,
+
     /// Key press.
     Key(KeyEvent),
+
     /// Mouse click/scroll.
     Mouse(MouseEvent),
+
     /// Terminal resize.
+    ///
+    /// This event may be fired multiple times in quick succession, so it is recommended not to
+    /// render the UI on every resize event.
     Resize(u16, u16),
+
+    /// An error occurred.
+    Error(std::io::Error),
 }
 
 /// Terminal event handler.
-#[allow(dead_code)]
+///
+/// This struct is responsible for listening to terminal events and sending them to the main thread.
+///
+/// It sends a tick event at a regular interval. The tick rate is specified by the `tick_rate`
+/// parameter. If an error occurs, it sends an error event to the main thread and then stops
+/// running.
 #[derive(Debug)]
-pub struct EventHandler {
-    /// Event sender channel.
-    sender: mpsc::Sender<Event>,
+pub struct EventSource {
     /// Event receiver channel.
     receiver: mpsc::Receiver<Event>,
-    /// Event handler thread.
-    handler: thread::JoinHandle<()>,
```

Comment:
Leaving this in would be the way of saying "hey you can actually access the event handler thread" like this.

Reply:
Can you give a concrete example of why a simple app would need to do that?

My goal with this template is to make it the sort of thing you'd reach for instead of cargo new. It should have the barest of things and should embody the YAGNI spirit. Another way of looking at this is to choose features where there is a demonstrated need, that need can be easily documented and supported by code that achieves this. Here it would be rare to need this.

Reply:
This is needed for killing the thread when you want to spawn vim, and need to spawn a new thread.

I agree about YAGNI spirit but I also would like to have it set up such that it extends well to future use cases. For example, when learning a new library or application, if I start with a simple tutorial example and find that it doesn't scale well or doesn't work with a feature I want to implement, I personally find that annoying. I do think it is quite common though.